elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -0,0 +1,98 @@
+import com.yahoo.elide.datastores.aggregation.metric.Aggregation;
+import com.yahoo.elide.datastores.aggregation.metric.Metric;
+import com.yahoo.elide.datastores.aggregation.schema.Schema;
+import com.yahoo.elide.request.Attribute;
+import com.yahoo.elide.request.EntityProjection;
+import com.yahoo.elide.request.Relationship;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class AggregationDataStoreHelper {
+
+    private Schema schema;
+    private EntityProjection entityProjection;
+
+    public AggregationDataStoreHelper(Schema schema, EntityProjection entityProjection) {
+        this.schema = schema;
+        this.entityProjection = entityProjection;
+    }
+
+    public FilterExpression getWhereFilter(FilterExpression filterExpression) {
+        FilterExpression whereFilter;
+        try {
+            whereFilter = filterExpression.accept(new SplitFilterExpressionVisitor(schema)).getWhereExpression();
+        } catch (NullPointerException npe) {
+            whereFilter = null;
+        }
+        return whereFilter;
+    }
+
+    public FilterExpression getHavingFilter(FilterExpression filterExpression) {
+        FilterExpression havingFilter;
+        try {
+            havingFilter = filterExpression.accept(new SplitFilterExpressionVisitor(schema)).getHavingExpression();
+        } catch (NullPointerException npe) {
+            havingFilter = null;
+        }
+        return havingFilter;
+    }
+
+
+    public void populateDimensionList(Set<Dimension> dimensions, Set<TimeDimension> timeDimensions) {
+        Set<String> dimensionNames = getDimensionNames();
+        Set<String> metricNames = getMetricNames(); // time dimensions are under attribute in entity projection
+        for (Dimension dimension : schema.getDimensions()) {
+            if (dimension instanceof TimeDimension) {
+                if(metricNames.contains(dimension.getName()))
+                timeDimensions.add((TimeDimension)dimension);
+            }
+            else if (dimensionNames.contains(dimension.getName()) || metricNames.contains(dimension.getName())) {
+                dimensions.add(dimension);
+            }
+        }
+    }
+
+    public void populateMetricList(Set<Metric> metrics) {
+        Set<String> metricNames = getMetricNames();
+        for (Metric metric : schema.getMetrics()) {
+            if (metricNames.contains(metric.getName())) {
+                metrics.add(metric);
+            }
+        }
+    }
+
+    public Map<Metric, Class<? extends Aggregation>> getMetricMap(Set<Metric> metrics) {
+        Map<Metric, Class<? extends Aggregation>> metricMap = new LinkedHashMap<>();
+        for (Metric metric : metrics) {
+            metricMap.put(metric, metric.getAggregations().get(0));
+        }
+        return metricMap;
+    }
+
+    private Set<String> getMetricNames() {
```

Comment:
Instead of saying getMetricNames, let's call this getAttributes.  An attribute may be a metric, or a time dimension, or a degenerate dimension.  So getMetricNames makes the calling code look wrong.